### PR TITLE
feat(campaigns): add Android support for campaigns

### PR DESCRIPTION
**Pull request by kioli**
### reviewers
@tzengerink @Rifinio @kioli @SuzettePaeta @gPinato

### CODE REVIEW

File: usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java [hunk 1]
```diff
@@ -102,9 +101,34 @@ public void showLoadedFrom() {
...
     }
 
+    @ReactMethod
+    public void sendEvent(@NonNull final String eventName) {
+        final Activity activity = getCurrentActivity();
+        if (activity != null) {
+            Usabilla.sendEvent(activity.getBaseContext(), eventName);
+            return;
+        }
+        Log.e(LOG_TAG, "Sending event to Usabilla not possible. Android activity is null");
```
Comment (SuzettePaeta) [inline usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java]: missing "is" before not

File: usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java [hunk 2]
```diff
@@ -102,9 +101,34 @@ public void showLoadedFrom() {
...
+    }
+
+    @ReactMethod
+    public void resetCampaigns() {
+        final Activity activity = getCurrentActivity();
+        if (activity != null) {
+            Usabilla.resetCampaignData(activity.getBaseContext());
+            return;
+        }
+        Log.e(LOG_TAG, "Resetting Usabilla campaigns not possible. Android activity is null");
```
Comment (SuzettePaeta) [inline usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java]: same as above

File: App.js [hunk 3]
```diff
@@ -28,13 +28,23 @@ export default class App extends Component<{}> {
   requestForm() {
     usabilla.loadFeedbackForm("YOUR_FORM_ID_HERE")
   }
+  resetCampaigns() {
+    usabilla.resetCampaigns()
+  }
```
Comment (SuzettePaeta) [inline App.js]: missing new lines between methods?

Comment (kioli) [reply App.js]: So far we kept them one after the other without spaces

File: usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java (hunk 2 above)
Comment (kioli) [reply usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java]: 👍

File: usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java (hunk 1 above)
Comment (kioli) [reply usabilla-react-native/android/src/main/java/com/usabilla/reactnative/ubform/UsabillaBridge.java]: 👍

File: App.js (hunk 3 above)
Comment (SuzettePaeta) [reply App.js]: is it a js convention?

Comment (kioli) [reply App.js]: not really... checking other projects it seems they have spaces so spaces will be added

File: usabilla-react-native/src/index.js [hunk 4]
```diff
@@ -18,10 +18,18 @@ function loadFeedbackForm(formId) {
...
-function showLoadedForm(event) {
+function showLoadedForm() {
     UsabillaBridge.showLoadedFrom()
 }
 
+function sendEvent(event) {
+    UsabillaBridge.sendEvent(event)
+}
+
+function resetCampaigns() {
```
Comment (Rifinio) [inline usabilla-react-native/src/index.js]: I think that we should keep the same naming as in our Native SDKs and call this method `Usabilla.resetCampaignData`

Comment (kioli) [reply usabilla-react-native/src/index.js]: 👍

File: App.js [hunk 5]
```diff
@@ -22,19 +22,34 @@ export default class App extends Component<{}> {
...
   requestForm() {
     usabilla.loadFeedbackForm("YOUR_FORM_ID_HERE")
   }
+
+  resetCampaignData() {
+    usabilla.resetCampaignData()
+  }
+
+  sendEvent() {
+    usabilla.sendEvent("YOUR_EVENT_HERE")
```
Comment (Rifinio) [inline App.js]: Can we add a text field to the view that can allow us to send different events each time?

File: App.js [hunk 6]
```diff
@@ -17,24 +18,43 @@ const instructions = Platform.select({
...
       <View style={styles.container}>
         <Text style={styles.welcome}>Welcome to React Native!</Text>
         <Text style={styles.instructions}>To get started, edit App.js</Text>
         <Text style={styles.instructions}>{instructions}</Text>
         <Button onPress={this.requestForm} title="Form" />
+        <TextInput
+          style={{height: 40, width: 150}}
+          placeholder="Event"
+          onChangeText={(text) => this.setState({text})} />
+        <Button onPress={() => this.sendEvent(this.state.text)} title="Campaign" />
```
Comment (Rifinio) [inline App.js]: the title of the button should be `Send event` i guess

**Pull request merged**